SerialPrograms/Source/CommonFramework/Exceptions/UnexpectedBattleException.cpp

```diff
@@ -0,0 +1,63 @@
+/*  Unexpected Battle Exception
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "CommonFramework/ImageTypes/ImageRGB32.h"
+#include "CommonFramework/Notifications/ProgramNotifications.h"
+#include "CommonFramework/Tools/ErrorDumper.h"
+#include "CommonFramework/Tools/ProgramEnvironment.h"
+#include "CommonFramework/Tools/ConsoleHandle.h"
+#include "UnexpectedBattleException.h"
+
+namespace PokemonAutomation{
+
+
+UnexpectedBattleException::UnexpectedBattleException(ErrorReport error_report, Logger& logger, std::string message)
```

Comment:
With this, there will be 4 exception classes with near identical implementations. I'll probably revisit this in future to try to merge away some of this duplicate code.